DFC.Composite.Shell.Services/UriSpecifcHttpClient/UriSpecifcHttpClientFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+
+namespace DFC.Composite.Shell.Services.UriSpecifcHttpClient
+{
+    public class UriSpecifcHttpClientFactory : IUriSpecifcHttpClientFactory
+    {
+        private readonly IHttpClientFactory httpClientFactory;
+        private readonly List<string> registeredUrls;
+
+        public UriSpecifcHttpClientFactory(IHttpClientFactory httpClientFactory, IRegisteredUrls registeredUrls)
+        {
+            this.httpClientFactory = httpClientFactory;
+            this.registeredUrls = registeredUrls?.GetAll().Select(url => $"{url}_{nameof(UriSpecifcHttpClientFactory)}").ToList();
+        }
+
+        public HttpClient GetClientForRegionEndpoint(string url)
+        {
+            var key = $"{url}_{nameof(UriSpecifcHttpClientFactory)}";
```

Comment:
We can use named clients with IHttpClientFactory instead of writing this logic I suppose. See this: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0#named-clients

Reply:
Mine is ultimately a wrapper over that, yeah.

The bit I don't think we can do without this, is pointing new pages that are added during the lifetime of the instance of shell - we need to direct them to a named instance (the default one) rather then to a new one, as we set up a bunch of stuff on the named ones

Reply:
After discussion, we determined it was only new CUI apps it would matter for so, so we removed the extra code and simplified